Add "Why Rhino?" and clean up main readme (#234)
Appsilon/rhino: vignettes/explanation-what-is-rhino.Rmd

# Why not use Golem, Leprechaun?
Golem is a great package, and we believe it is a good fit for many developers. Choose whichever works best for you and your project!

> You can learn more about the differences in [How is Rhino different from ...?](articles/about-how-is-it-different.html).

Reviewer: ```suggestion
You can learn more about the differences in [Similar projects](articles/explanation-similar-projects.html).
```
